cdap-ui/app/cdap/components/FastAction/FastActionButton/index.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {PropTypes} from 'react';
+
+export default function FastActionButton({icon, api, params, onSuccess, onError, onClick}) {
+  function clickHandler() {
+    if (typeof onClick === 'function') {
+      onClick();
+    }
+
+    api(params)
+      .subscribe(onSuccess, onError);
+  }
+
+  return (
+    <button
+      className="btn btn-link"
+      onClick={clickHandler}
+    >
+      <span className={icon}></span>
+    </button>
+  );
+}
+
+
+FastActionButton.propTypes = {
+  icon: PropTypes.string,
+  api: PropTypes.func,
+  params: PropTypes.object,
```

Comment:
If we can define the list of fast action types then can't each fastaction type have a specific param? I'm not sure what I am overseeing here but any specific reason why its not an option?